sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,182 @@
+                    }
+
+                    // Create and return a ChatThreadClient for the new ChatThread
+                    let chatThreadClient = try ChatThreadClient(
+                        endpoint: self.endpoint,
+                        credential: self.credential,
+                        threadId: threadId,
+                        withOptions: self.options
+                    )
+
+                    completionHandler(.success(chatThreadClient), httpResponse)
+
+                case let .failure(error):
+                    throw error
+                }
+            } catch {
+                let azureError = AzureError.client("Failed to create ChatThread.", error)
+                completionHandler(.failure(azureError), httpResponse)
+            }
+        }
+    }
+
+    /// Get the ChatThread with given id.
+    /// - Parameters:
+    ///   - withId: The chat thread id.
+    ///   - withOptions: Get chat thread options.
+    ///   - completionHandler: A completion handler that receives the chat thread on success.
+    public func getChatThread(
+        withId chatThreadId: String,
+        withOptions options: AzureCommunicationChatService.GetChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatThread>
+    ) {
+        service.getChatThread(chatThreadId: chatThreadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(chatThread):
+                completionHandler(.success(chatThread), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Gets the list of ChatThreads for the user.
+    /// - Parameters:
+    ///   - withOptions: List chat threads options.
+    ///   - completionHandler: A completion handler that receives the list of chat thread info on success.
+    public func listChatThreads(
+        withOptions options: AzureCommunicationChatService.ListChatThreadsOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<ChatThreadInfo>>
+    ) {
+        service.listChatThreads(withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(chatThreads):
+                completionHandler(.success(chatThreads), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Delete the ChatThread with id chatThreadId.
+    /// - Parameters:
+    ///   - chatThreadId: The chat thread id.
+    ///   - withOptions: Delete chat thread options.
+    ///   - completionHandler: A completion handler.
+    public func delete(
+        chatThreadId: String,
+        withOptions options: AzureCommunicationChatService.DeleteChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
```

Comment:
This is now inconsistent with `getChatThread(withId: ...)`. Whatever naming convention is used, it should be consistent across the SDK.